src/components/course/data/hooks.jsx

```diff
@@ -807,41 +804,37 @@ export const useUserSubsidyApplicableToCourse = ({
         if (legacyUserSubsidyApplicableToCourse) {
           // check for exceeded remaining spend/enrollments and per-learner limits if it's an enterprise offer
           if (legacyUserSubsidyApplicableToCourse.subsidyType === ENTERPRISE_OFFER_SUBSIDY_TYPE) {
-            const {
-              remainingBalance,
-              remainingBalanceForUser,
-              remainingApplications,
-              remainingApplicationsForUser,
-            } = legacyUserSubsidyApplicableToCourse;
-
-            const hasBalanceRemainingForUser = (
-              !isNil(remainingBalanceForUser) ? remainingBalanceForUser > courseListPrice : true
-            );
-            const hasRemainingApplicationsForUser = (
-              !isNil(remainingApplicationsForUser) ? remainingApplicationsForUser > 0 : true
-            );
-            const hasBalanceRemaining = !isNil(remainingBalance) ? remainingBalance > courseListPrice : true;
-            const hasRemainingApplications = !isNil(remainingApplications) ? remainingApplications > 0 : true;
-
-            const redeemableOfferConditions = [
-              hasBalanceRemainingForUser,
-              hasRemainingApplicationsForUser,
-              hasBalanceRemaining,
-              hasRemainingApplications,
-            ];
-
-            if (redeemableOfferConditions.every(offerCondition => offerCondition)) {
+            const redeemableOffer = isOfferRedeemableForCourse({
+              offer: legacyUserSubsidyApplicableToCourse,
+              coursePrice: courseListPrice,
+            });
+
+            if (redeemableOffer.resolve) {
               // Redeemable for this course
               setUserSubsidyApplicableToCourse(legacyUserSubsidyApplicableToCourse);
               setMissingUserSubsidyReason(undefined);
             } else {
-              let ineligibleEnterpriseOfferReasonType = DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY;
+              const {
+                hasRemainingApplicationsForUser,
+                hasRemainingBalanceForUser,
+                hasRemainingBalance,
+                hasCurrent,
+              } = redeemableOffer.conditions;
```

Comment:
[clarification] It appears the `conditions` returned by `isOfferRedeemableForCourse` is an array of booleans. However, here we're destructuring `conditions` as if it's an object with key/value pairs?

Should the helper function that generates these values be returning them as an object?

Reply:
The helper, `.conditions`, does return as an object, but not all returned values from the default object are required. `hasRemainingApplications` key is not being used in `hooks.js` and it would require referencing the new variable at each conditional if destructuring wasn't used. 

It was an intentional use of destructuring.

Reply:
Ended up being a logical error on my part, should of been wrapped in a object with some additional fixes downstream to handle the parsing of an array.
Thanks for the huddle to call this out 👍🏽